[RFC] docs: Mention differences between vim and nvim

Pyrohh opened the pull request:
It's a little annoying not having highlighting for breakindent, so I just added it manually.

fwalch commented:
I think this is in Vim patch https://code.google.com/p/vim/source/detail?r=b3bc99b909c3

Commit:
doc: Introduce vim-differences.txt

This documents the differences between nvim and nvim.

Regarding the removal of references to 'renderoptions': it was never
added in the first place, so there's no need to mention its "removal".

Pyrohh commented:
@fwalch indeed, I dropped my syntax changes.

justinmk commented on runtime/doc/options.txt:
```diff
@@ -5560,7 +5560,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 
 						*'renderoptions'* *'rop'*
-'renderoptions' 'rop'	Removed. {Nvim} will choose the best renderer available.
+'renderoptions' 'rop'	Removed. {Nvim}
```
We added this line intentionally to communicate that Neovim doesn't need this option. Maybe the wording can be improved, idk.

Pyrohh replied:
I think adding reasoning for removal is a good idea, but not here. This is the only removed option with a description, and it's incomplete. Perhaps we can just replace `vi_diff.txt` with `nvim_diff.txt`, then tack on `(See |nvim-diff-options|)` to the end of each removed option in `options.txt`.

I can convert this PR, or just open a new one for a cleaner comment history.

justinmk replied:
> Perhaps we can just replace vi_diff.txt

Let's leave vi_diff.txt alone for now, but create a new `vim_diff.txt` file. 

> then tack on (See |nvim-diff-options|)

I suggest `|vim-differences|` in analogy with the existing `vi-differences` tag.

> nvim_diff.txt

Technically it should be `vim_diff.txt` :)

> I can convert this PR,

:+1:

Pyrohh commented:
Is it possible to have @coveralls ignore PRs with `docs:` in the title?

fwalch commented:
@justinmk Can you check the Coveralls settings ("leave comments" should be off, "use status API" should be on, I guess). https://github.com/lemurheavy/coveralls-public/issues/25#issuecomment-72534212

@Pyrohh Hm, I don't think this info is available on Travis by default, but we could use the Github API.. not sure if that's worth the effort, though. If it's too annoying, we could just turn it off again.

Pyrohh commented:
Thanks @kopischke, repushed.

Pyrohh commented:
@fwalch When you say Travis do you mean coveralls? If so, I don't see any reason to turn it off assuming we instead use the status API like you mentioned (which I wasn't aware about).

fwalch commented:
@Pyrohh I thought that we could just not upload coverage data from the Travis scripts, then we also wouldn't get a notification in the PR.

Pyrohh commented:
> I thought that we could just not upload coverage data from the Travis scripts, then we also wouldn't get a notification in the PR.

I'm a little confused, are you saying that Coveralls is built in to Travis? I'm under the assumption that Coveralls is a paid service, separate from Travis. So when a new commit lands in a PR, Github's API catches that, and sends two messages to two separate services.

fwalch commented:
Yes, coveralls is a separate service, but the tests are still run on Travis, only the coverage data is uploaded. I believe it is also supported by Travis, but the way we do it is to upload the coverage results manually after we run the tests. Look for `coveralls` at the end of the scripts in`.ci`. This then causes coveralls to analyze this data, which will result in a comment/build status update in the Github issue.

Pyrohh commented:
Ah, thanks for the explanation. We might as well try it, I can't imagine it'd be too hard.

Pyrohh commented:
Ah, thanks for the explanation. We might as well try it, I can't imagine it'd be too hard.

justinmk commented:
@fwalch I turned off the coveralls comments. Reason I enabled it originally is that I do not see any effect with just "use status API".

fwalch commented:
> @fwalch I turned off the coveralls comments. Reason I enabled it originally is that I do not see any effect with just "use status API".

Strange; let's see if it works this time..

@Pyrohh It seems to fit in this PR somewhat; can you address https://github.com/neovim/neovim/pull/1926#discussion_r23934196? What about adding a section on "new features", listing small ones such as the "EndOfBuffer" highlight group and linking to the help topics of the "big" ones (msgpack-rpc, providers, ...)?

Pyrohh commented:
@fwalch I repushed with a few small changes, but I'm not that content with the current structure. See my comments on the commit.

@justinmk do we still need client server refs (`remote.txt`)? I could do that in a different PR.

justinmk commented:
We will be restoring some clientserver features.

Pyrohh commented:
Repushed and setting as RFC.

Pyrohh commented:
Repushed due to #1945

justinmk commented on runtime/doc/options.txt:
```diff
@@ -4379,7 +4380,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 
 					*'key'*
-'key'			Removed. {Nvim}
+'key'			Removed; see |vim-differences|.
```
it's customary in vim help to just plop a tag right after a word, without using extra verbiage such as "see". (There are some exceptions). So how about this:

    'key'			Removed. |vim-differences| {Nvim}

Pyrohh replied:
Sounds good to me.

justinmk commented on runtime/doc/vim_diff.txt:
```diff
@@ -0,0 +1,96 @@
+
+- MS-DOS support
+  - 'bioskey'
+  - 'conskey'
```
I think the bullets might get a bit ugly. How do you feel about just indenting?

Pyrohh replied:
I reduced the indentation level by two spaces, so top-most items like `8.3 filesystem support` and `Encryption support` now lack bullets. I'm not sure if this is what you wanted, but IMO they make sense when listing off individual points.

Pyrohh commented:
Repushed with recommended changes, and added in a missing reference to `renderoptions`.

Pyrohh commented:
(sorry for flooding your mail) #1925 should probably be merged first, so the `compatible` reference in `options.txt` can be updated to follow the other removed options.

justinmk commented:
@Pyrohh BTW I noticed there are some newlines missing in the `nvim -h` output. Can you fix that in one of these existing PRs?

Pyrohh commented:
Sure, but where are they?

justinmk commented:
Last 3 lines... can't miss it :)

Pyrohh commented:
This shows up for me:
```
  --embed               Use stdin/stdout as a msgpack-rpc channel
  --version             Print version information and exit
  -h | --help           Print this help message and exit
```

justinmk commented:
My mistake, I was using an older version. Juggling too many systems... sorry.

Pyrohh commented:
That's fine.

Pyrohh commented:
Repushed after #1925

Pyrohh commented:
Hmm.. oldtest 61 is failing on OS X.

fwalch commented:
> Hmm.. oldtest 61 is failing on OS X.

Same for my PR (#2002).

Pyrohh commented:
This is all ready on my end, but I have no preference whether this or #2008 get merged first. One thing though: I noticed that the new `Tab*` events don't have an `{Nvim}` tag next to them in the docs, should that be added? I didn't already because I saw they were sent to vim_dev.

fmoralesc commented:
They should probably be added while vim_dev doesn't merge them. Bram seemed alright with the patches, but who knows when that'll happen.

Pyrohh commented:
Alright, I added the tags.

Pyrohh commented:
Setting back to RFC, the vim differences wiki page has been updated a lot since so I'll try to sync this PR with that. Also it should be easier to merge this after #2008, as to document said PRs changes while avoiding a chicken-egg problem.

Pyrohh commented:
I've updated this for #2008, and closed in favor of #2141.
